red/Animal/scripts/barnyard.pl

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/perl
+
+use v5.14;
+use strict;
+use warnings;
+
+use Cow;
+use Horse;
+use Sheep;
+use Mouse;
+
+our @ANIMALS = qw( Cow Horse Sheep Mouse );
+
+print "Enter animal names to gain a deep understanding of how they communicate.\n";
+
+my @choice;
+while ( <STDIN> ) {
+  chomp and my $animal = $_;
+
+  if ( $animal ~~ @ANIMALS ) {
```

Comment:
It's pretty neat, though I'd probably normally use a set to avoid O(n) comparisons. Something like `our %ANIMALS = map $_ => 1, qw( Cow Horse Sheep Mouse );`.